externals/kyuubi-trino-engine/src/test/scala/org/apache/kyuubi/engine/trino/operation/TrinoOperationSuite.scala

```diff
@@ -45,6 +46,61 @@ class TrinoOperationSuite extends WithTrinoEngine with HiveJDBCTestHelper {
 
   override protected def jdbcUrl: String = getJdbcUrl
 
+  test("trino - get type info") {
+    withJdbcStatement() { statement =>
+      val typeInfo = statement.getConnection.getMetaData.getTypeInfo
+      val types = Set(
+        BIGINT,
+        INTEGER,
+        SMALLINT,
+        TINYINT,
+        BOOLEAN,
+        DATE,
+        DECIMAL,
+        REAL,
+        DOUBLE,
+        HYPER_LOG_LOG,
+        QDIGEST,
+        P4_HYPER_LOG_LOG,
+        INTERVAL_DAY_TO_SECOND,
+        INTERVAL_YEAR_TO_MONTH,
+        TIMESTAMP,
+        TIMESTAMP_WITH_TIME_ZONE,
+        TIME,
+        TIME_WITH_TIME_ZONE,
+        VARBINARY,
+        VARCHAR,
+        CHAR,
+        ROW,
+        ARRAY,
+        MAP,
+        JSON,
+        IPADDRESS,
+        UUID,
+        GEOMETRY,
+        SPHERICAL_GEOGRAPHY,
+        BING_TILE,
+        "color",
+        "KdbTree",
+        "CodePoints",
+        "JsonPath",
+        "Regressor",
+        "JoniRegExp",
+        "unknown",
+        "ObjectId",
+        "SetDigest",
+        "Re2JRegExp",
+        "Model",
+        "tdigest",
+        "LikePattern",
+        "function",
+        "Classifier")
+      while (typeInfo.next()) {
+        assert(types.contains(typeInfo.getString(TYPE_NAME)))
+      }
```

Comment:
Shall we add a check for the total type size ? It would be helpful to detect the implicit change.
```scala
while (typeInfo.next()) {
  assert(types.contains(typeInfo.getString(TYPE_NAME)))
  i += 1
}
assert(types.size == i)
```